src/pf/parse.lua

```diff
@@ -24,12 +24,55 @@ local punctuation = set(
    '+', '-', '*', '/', '&', '|', '^', '&&', '||', '<<', '>>', '\\'
 )
 
+local number_terminators = " \t\r\n)]:!<>=+-*/%&|^"
+
+local function lex_number(str, pos, base)
+   local res = 0
+   local i = pos
+   while i <= #str do
+      local chr = str:sub(i,i)
+      local n = tonumber(chr, base)
+      if n then
+         res = res * base + n
+         assert(res <= 0xffffffff, 'integer too large: '..res)
```

Comment:
Is 0xffffffffffoo.com a valid hostname?  I would think so.  Probably want to move this assertion to the end, when we know we've seen a terminator and have a valid number.  WDYT?